src/test/diagnostics_test.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+import * as path from 'path';
+import {ResolvedUrl} from 'polymer-analyzer/lib/model/url';
+import {Diagnostic} from 'vscode-languageserver-types/lib/main';
+import {DiagnosticSeverity} from 'vscode-languageserver/lib/main';
+
+import {assertDoesNotSettle, createTestEnvironment} from './util';
+
+const fixtureDir = path.join(__dirname, '..', '..', 'src', 'test', 'static');
+
+suite('DiagnosticGenerator', function() {
+  const indexPath = path.join('editor-service', 'index.html') as ResolvedUrl;
+
+  test('For a good document we get no warnings', async() => {
```

Comment:
nit: could you use the same code style for test names, aka have here on top `let testName = '...'` and reuse the variable for the following tests? or remove the usages of `let testName` below, either is fine